packages/override_api_endpoint/lib/src/override_api_endpoint.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:uni_links/uni_links.dart';
+
+Future<Uri> overrideApiEndpoint(
+  String deeplinkOverrideSegment,
+  String deeplinkQueryParameter,
+  Uri defaultEndpoint,
+) async {
+  const apiEndpointKey = 'apiAddress';
+
+  final sharedPreferences = await SharedPreferences.getInstance();
+
+  var apiEndpoint = defaultEndpoint;
+
+  if (sharedPreferences.containsKey(apiEndpointKey)) {
+    apiEndpoint = Uri.parse(sharedPreferences.getString(apiEndpointKey)!);
+  }
+
+  final initial = await getInitialUri();
+
+  if (initial != null && initial.path.contains(deeplinkOverrideSegment)) {
+    final endpointFromDeeplink =
+        initial.queryParameters[deeplinkQueryParameter];
+    if (endpointFromDeeplink?.isEmpty ?? true) {
+      sharedPreferences.remove(apiEndpointKey);
+      apiEndpoint = defaultEndpoint;
+    } else {
+      apiEndpoint = Uri.parse(endpointFromDeeplink!);
+      sharedPreferences.setString(apiEndpointKey, apiEndpoint.toString());
```

Comment:
Why not simply using the `endpointFromDeeplink` instead of the stringified `apiEndpoint`?

Reply:
The main idea was to return the apiEndpoint at the end and persist it if necessary. Thus I wrote it this way

Reply:
Let's resolve this conversation